test/unit/FutarchyChallenge.js

```diff
@@ -0,0 +1,259 @@
+    })
+  })
+
+  describe('ended()', async () => {
+    describe('contingent on futarchOracle outcome', async () => {
+      beforeEach(async () => {
+        challenge = await deployChallenge()
+        await challenge.start()
+        futarchyOracleMock = await FutarchyOracleMock.at(await challenge.futarchyOracle())
+      })
+
+      it('returns true if FutarchyOracle outcome is set', async () => {
+        await futarchyOracleMock.mock_setIsSet(true)
+        expect(await challenge.ended()).to.equal(true)
+      })
+
+      it('returns false if FutarchyOracle outcome is not set', async () => {
+        await futarchyOracleMock.mock_setIsSet(false)
+        expect(await challenge.ended()).to.equal(false)
+      })
+    })
+
+    // TODO: Update tests on fund() / start() conditions when that logic is updated
+  })
+
+  describe('passed()', async () => {
+    beforeEach(async () => {
+      challenge = await deployChallenge()
+      await challenge.start()
+      futarchyOracleMock = await FutarchyOracleMock.at(await challenge.futarchyOracle())
+    })
+
+    it('reverts if ended() returns false', async () => {
+      await futarchyOracleMock.mock_setIsSet(false)
+      await expectRevert(challenge.passed())
+    })
+
+    it('returns true if futarchyOracle outcome equals 1', async () => {
+      // winningIndex 1 = scalarDenied, meaning listing is denied and FutarchyChallenge passes
+      await futarchyOracleMock.mock_setIsSet(true)
+      await futarchyOracleMock.mock_setWinningMarketIndex(1)
+      expect(await challenge.passed()).to.equal(true)
+    })
+
+    it('returns false if futarchyOracle outcome equals 0', async () => {
+      // winningIndex 0 = scalarAccepted, meaning listing is accepted and FutarchyChallenge fails
+      await futarchyOracleMock.mock_setIsSet(true)
+      await futarchyOracleMock.mock_setWinningMarketIndex(0)
+      expect(await challenge.passed()).to.equal(false)
+    })
+  })
+
+  describe('winnerRewardAmount()', async () => {
+    let winnerRewardAmount;
+
+    beforeEach(async () => {
+      token = await Token.new(1000 * 10 ** 18, "FCR Token", 18, 'FCR')
+      challenge = await deployChallenge({token})
+      await challenge.start()
+      winnerRewardAmount = 2 * 10 ** 18
+      await token.transfer(challenge.address, winnerRewardAmount)
+    })
+
+    it('reverts if marketsAreClosed is false', async () => {
+      expectRevert(challenge.winnerRewardAmount());
+    })
+
+    it('returns the value of winnerRewardAmount', async () => {
+      await challenge.close()
+      expect((await challenge.winnerRewardAmount()).toNumber()).to.equal(winnerRewardAmount)
+    })
+
+    it('returns contract balance at the time close() is invoked even if the contract token balance since increased', async () => {
+      // if winnerRewardAmount is larger than what the challenge has approved to the registry in close(),
+      // tokens will be forever locked
```

Comment:
Sounds maybe undesirable for the tokens to be locked forever.  Realize this is not exactly relevant to this test, but could this be prevented somehow?

Reply:
I think this has more to do with the erc20 standard than this contract...you can transfer a token to any address you'd like, so send it to the wrong address....you've lost your money.

With that, I had a bug where I approved the current challenge balance to the registry, then the registry can later try and `transferFrom` the current balance. If that balance has since went up, there wouldn't be enough approved tokens, so I had to assign that number to a variable so it was permanent.

Reply:
Ill look into the wording of this comment to see if I can clarify what I'm trying to do better